src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/matmul.cpp

```diff
@@ -120,11 +120,13 @@ void MatMulLayerCPUTest::SetUp() {
     auto it = additionalConfig.find(ov::hint::inference_precision.name());
     if (it != additionalConfig.end() && it->second.as<ov::element::Type>() == ov::element::bf16)
         inType = outType = netType = ElementType::bf16;
+    else if (it != additionalConfig.end() && it->second.as<ov::element::Type>() == ov::element::f16)
+        inType = outType = netType = ElementType::f16;
     else
         inType = outType = netType;
 
     cpuNodeType = nodeType == MatMulNodeType::MatMul ? "MatMul" : "FullyConnected";
-    selectedType = makeSelectedTypeStr(selectedType, outType);
+    selectedType = makeSelectedTypeStr(selectedType, get_default_imp_precision_type(outType));
```

Comment:
why not reuse `CPUTestsBase::updateSelectedType`? I saw some single layer tests (eltwise/reduce/mvn) uses this function to handle inferen-precision hint

Reply:
I found CPUTestsBase::updateSelectedType is good function, but it doesn't think about hardware,
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/test_utils/cpu_test_utils.cpp#L391 for example , if net_type is bf16, but the hardware do not support it, the net_type will became to f32, maybe I can modify the function and reuse it